pkg/webhooks/hiveextension/v1beta1/agentclusterinstall_mutating_hook.go

```diff
@@ -150,19 +155,30 @@ func (a *AgentClusterInstallMutatingAdmissionHook) SetDefaults(admissionSpec *ad
 	}
 }
 
+func isNonePlatformOrSNO(newObject *hiveext.AgentClusterInstall) bool {
+	return isSNO(newObject) && (newObject.Spec.PlatformType == "" || newObject.Spec.PlatformType == hiveext.NonePlatformType) ||
+		newObject.Spec.PlatformType == hiveext.NonePlatformType
+}
+
 func patchUserManagedNetworking(newObject *hiveext.AgentClusterInstall, logger *log.Entry) map[string]interface{} {
-	var message string
-	var userManagedNetworking bool
-
-	if isSNO(newObject) {
-		message = "setting UserManagedNetworking to true with SNO"
-		userManagedNetworking = true
-	} else {
-		message = "setting UserManagedNetworking to false"
-		userManagedNetworking = false
+	var (
+		platformUserManagedNetworking *bool
+		userManagedNetworking         = isNonePlatformOrSNO(newObject)
+		highAvailabilityMode          = getHighAvailabilityMode(newObject, nil)
+		err                           error
+	)
+
+	if !isNonePlatformOrSNO(newObject) && newObject.Spec.PlatformType != "" {
+		platform := common.PlatformTypeToPlatform(newObject.Spec.PlatformType)
+		_, platformUserManagedNetworking, err = provider.GetClusterPlatformByHighAvailabilityMode(platform, nil, highAvailabilityMode)
+		if err != nil {
+			logger.Warnf("Cannot set UserManagedNetworking automatically due to: %s", err.Error())
+			return nil
+		}
+		userManagedNetworking = swag.BoolValue(platformUserManagedNetworking)
```

Comment:
Not sure I follow.
In line 178, we set UMA based on what the backend calculated. This calculation overrides the previous calculation as the case of a specified platform (which is not `None`) + none SNO cluster takes precedence.

Reply:
Yeah, I've just meant that it's a bit confusing that we set 'userManagedNetworking' twice in this case.
Maybe we can encapsulate it in a new getUserManagedNetworking func for example instead of overriding.
But can be handled later on.